examples/gno.land/r/demo/grc20factory/grc20factory.gno

```diff
@@ -1,140 +1,185 @@
-package foo20
+// Package grc20factory defines a factory realm for grc20 tokens.
+// It serves as a registry/proxy for accessing all tokens registered in it.
+
+package grc20factory
 
 import (
+	"bytes"
+	"math"
 	"std"
-	"strconv"
 	"strings"
 
 	"gno.land/p/demo/avl"
 	"gno.land/p/demo/grc/grc20"
+	"gno.land/p/demo/ufmt"
 )
 
-// XXX: the p/grc20 library needs a refactor to change names (i.e., adminToken)
-
-type token struct {
-	adminToken *grc20.AdminToken
-	admin      std.Address
+type tokenWithAdmin struct {
+	token    *grc20.Token
+	admin    std.Address
+	fauceted bool
 }
 
-var tokens avl.Tree // symbol -> token
+var tokens avl.Tree // symbol -> tokenWithAdmin
 
-func NewToken(name, symbol string, decimals uint, initialMint uint64) {
+func NewToken(name, symbol string, decimals uint8, initialMint uint64) {
 	admin := std.PrevRealm().Addr()
 	NewTokenWithAdmin(name, symbol, decimals, initialMint, admin)
 }
 
-func NewTokenWithAdmin(name, symbol string, decimals uint, initialMint uint64, admin std.Address) {
+func NewTokenWithAdmin(name, symbol string, decimals uint8, initialMint uint64, admin std.Address) {
 	exists := tokens.Has(symbol)
 	if exists {
-		panic("token already exists")
+		err := ufmt.Sprintf("token with symbol $%s already exists", symbol)
+		panic(err)
 	}
 
-	newToken := grc20.NewAdminToken(name, symbol, decimals)
+	newToken := grc20.NewGRC20Token(name, symbol, decimals)
 	newToken.Mint(admin, initialMint)
-	t := token{
-		adminToken: newToken,
-		admin:      admin,
+
+	t := tokenWithAdmin{
+		token:    newToken,
+		admin:    admin,
+		fauceted: false, // faucet disabled by default
```

Comment:
If disabled by default, please register one or two tokens in `init()` that will be fauceted as true by default, so this package can be easily tested.